samples/bluetooth/central_and_peripheral_hr/Kconfig.sysbuild

```diff
@@ -0,0 +1,11 @@
+#
+# Copyright (c) 2023 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+source "${ZEPHYR_BASE}/share/sysbuild/Kconfig"
+
+config NRF_DEFAULT_BLUETOOTH
```

Comment:
The idea of this is that it selects the features it needs, some applications need bluetooth, some applications need 802.15.4, some need both. There are multiple applications which can run on the network core, so if both bluetooth and 802.15.4 are required than this is prevent just showing the bluetooth or or just the 802.15.4 one, when it should only show the dual image option.

Reply:
I see, in case both Bluetooth and 802.15.4 are enabled we are eventually going to use the `hci_multiprotocol` I guess (which seems not yet supported with sysbuild). Maybe we can e.g. fail build or emit warning if we cannot support the needed features?
Also are the `NRF_DEFAULT_...` prefixed options defined somewhere (so user would be able to find a reference if needed)?

Reply:
This is also probably going to be replaced in the future with a system that can select the image automatically

Reply:
I am not sure about it - this is what I am doing for a few minutes - searching where NRF_DEFAULT_BLUETOOTH is defined and described. What we need is:
Somewhere in the generic Kconfig:
```
config NRF_DEFAULT_BLUETOOTH
   bool
   help
     This is an option to implement....
     Internal value set by the build system.
```

And then in a files that needs it:
```
config NRF_DEFAULT_BLUETOOTH
   default y
```

Reply:
The only generic Kconfig for sysbuild would be included for any project using NCS - even customer applications, this should not be visible or selectable for that.

Reply:
I'm not sure if the `NRF_DEFAULT_*` symbols should be promptless to begin with. If what this is building towards is indirect image selection based on features, then I would expect those features to be selected in `sysbuild.conf`:
```
SB_CONFIG_NRF_DEFAULT_BLUETOOTH=y
```
Samples already indicate that they need BLE or 802.15.4 using settings like `CONFIG_BT=y` or `CONFIG_ZIGBEE=y`.

Reply:
> The only generic Kconfig for sysbuild would be included for any project using NCS - even customer applications, this should not be visible or selectable for that.

but you are still referring to the symbol in the common sysbuild Kconfig code, like this:
https://github.com/nrfconnect/sdk-nrf/blob/a9bac1698b07c526d8f344ad5a5b0e0bcbadc387/sysbuild/Kconfig.netcore#L21

I agree that the symbol should be promptless, but I would still say that having a single definition as proposed here: https://github.com/nrfconnect/sdk-nrf/pull/11793/commits/49350e09e48232eee693b9005a0a77abbdfac53c#r1302914199 is valuable, especially because if done like that, then as soon as we remove
```
config NRF_DEFAULT_BLUETOOTH
   bool
   help
    ...
```
from common Kconfig, then this will issue a warning:
```
config NRF_DEFAULT_BLUETOOTH
   default y
```
making them easy to sport, and even more important, customers who follow our approach in custom app will immediately see the warning, whereas if we / they do the below in their sample, then no warning will be shown:
```
config NRF_DEFAULT_BLUETOOTH
	bool
	default y
```

See also: https://docs.zephyrproject.org/latest/build/kconfig/tips.html#common-kconfig-shorthands
> **Note**
>
> For a symbol defined in multiple locations (e.g., in a Kconfig.defconfig file in Zephyr), it is best to only give the symbol type for the “base” definition of the symbol, and to use default (instead of def_<type> value) for the remaining definitions. That way, if the base definition of the symbol is removed, the symbol ends up without a type, which generates a warning that points to the other definitions. That makes the extra definitions easier to discover and remove.

Reply:
> Samples already indicate that they need BLE or 802.15.4 using settings like CONFIG_BT=y or CONFIG_ZIGBEE=y.

Which cannot be supported in sysbuild, to know if one of these symbols is set, the main application needs to have been configured and Kconfig loaded into sysbuild, at which point the images have already been added and are being configured so it is too late

Reply:
> > Samples already indicate that they need BLE or 802.15.4 using settings like CONFIG_BT=y or CONFIG_ZIGBEE=y.
> 
> Which cannot be supported in sysbuild, to know if one of these symbols is set, the main application needs to have been configured and Kconfig loaded into sysbuild, at which point the images have already been added and are being configured so it is too late

I know. I wasn't suggesting to access this information before selecting the netcore image. I meant this as a supporting argument for why it could make sense for `NRF_DEFAULT_*` to have a prompt, i.e., as another feature selector but in sysbuild scope.

Reply:
Hi @nordicjm , could this maybe be enabled along with enabling BLE? That is to avoid repeatedly creating this file for every sample/application.

Reply:
No, a sysbuild file is required per application to select which additional images are used.

Reply:
>  is required per application to select which additional images are used

This selection is indirectly done by enabling BLE. That's how child images are doing it anyway. That's why I have asked.
If it is not possible, I understand.

Reply:
Child/parent image has the full application Kconfig loaded because that is the root project that configuration runs in, but sysbuild does not, it has only the sysbuild Kconfig tree loaded which is why it cannot automatically load images but child/parent images can.

Reply:
thanks for clarification